### ISSUE TITLE
Rename whitelist to something 

### ISSUE THREAD

**Issue by bitboxer** [mrdokenny/Cookie-AutoDelete#754]
In the face of all the current things going on in the world it would be a nice way of showing support by renaming graylist and whitelist to something else. You for sure will be able to give that concept another name that might not carry the co notation that the term whitelist carries with it.

Thanks ❤️ .

**Comment by kennethtran93** [comment]
While I'd like for this to be an instant change, there is a bit of coding to actually change the entire backend code that uses 'grey' and 'white' (not to mention the locales that will need to be translated on Crowdin as well).

In other words, it will be worked upon, but don't expect anything to be quickly published in the near future (next minor/patch version) or so.

**Comment by SolveEverything** [comment]
no whitelist is extremely standard, do not change that 

- it's be an extremely bad idea to change sometihng that is so standard, 
- it'd make this extension more confusing than it already is

this extension does not have good ux/ui design like other extensions and softwares do
- look at how simple + clean other things are, this extensions is so complicated 
- the only postiive is that the options has a link that goes to a github section explaining what something does (but it is without pictures like the good extensions have)

whitelist = allowing something that isnt blacklisted, allowing someting outside of the default
and whitelist will stay standard forever

naming of greylist should be changed cos it's extremely confusing, and nobody uses the term 'greylist'

**Comment by bitboxer** [comment]
Even if they were standard in the past it's time to change them. Companies like Microsoft already do this for a long time. see [here](https://www.theregister.com/2019/09/03/chromium_microsoft_offensive/) for an example of it. 

Hurtful language needs to stop. Language can change. To the better. It does not hurt you to use "Allowlist" instead. There is no harm done to you.

**Comment by Endymion101** [comment]
Please prioritize features that people actually care about. This comment has only received 3 votes, and yet you claim to support an "instant change" Complaints about unwanted notifications showing from "manual cleaning" have received over four times as many comments, and yet you say it's "not a big priority at this time?" Why not? How about prioritizing it?

**Comment by ghost** [comment]
> In the face of all the current things going on in the world it would be a nice way of showing support by renaming graylist and whitelist to something else. You for sure will be able to give that concept another name that might not carry the co notation that the term whitelist carries with it.
> 
> Thanks ❤️ .

Please prioritize features that people actually care about. This comment has only received 3 votes, and yet you claim to support an "instant change?" Complaints about unwanted notifications showing from "manual cleaning" have received over four times as many comments, and yet you say it's "not a big priority at this time?" Why not? How about prioritizing it?

**Comment by kennethtran93** [comment]
> > In the face of all the current things going on in the world it would be a nice way of showing support by renaming graylist and whitelist to something else. You for sure will be able to give that concept another name that might not carry the co notation that the term whitelist carries with it.
> > Thanks heart .
> 
> Please prioritize features that people actually care about. This comment has only received 3 votes, and yet you claim to support an "instant change?" Complaints about unwanted notifications showing from "manual cleaning" have received over four times as many comments, and yet you say it's "not a big priority at this time?" Why not? How about prioritizing it?

You seem to be mistaken about 'instant change' from my comment of 
> "While I'd like for this to be an instant change, there is a bit of coding to actually change the entire backend code that uses 'grey' and 'white' (not to mention the locales that will need to be translated on Crowdin as well)."

to which I implied it won't be easy.  I also have said 
> In other words, it will be worked upon, but don't expect anything to be quickly published in the near future (next minor/patch version) or so.

This can also be applied to the other issue about notifications as well...

BTW the more you keep making comments like these, the more time I need to read though and write up responses, which takes time away from what little I already have to code.  Keep that in mind please.

**Comment by Chaddock** [comment]
The terms have nothing to do with race.

-1

**Comment by AO-LocLab** [comment]
Whether terms like “white list” are acceptable or not is not left to the individual, the industry as a whole is considering it as a harmful term. 
Please read further about what these terms imply:
https://www.localizationlab.org/blog/2020/8/19/it-starts-with-words-unconscious-bias-in-gender-race-and-class-in-tech-terminology

If companies like IBM, Microsoft, Autodesk, etc. have decided to temporarily stop software development to prioritize the replacement of all these terms, “Cookie Autodelete” should consider to also make it a priority by respect to its users. I’m afraid that seeing these changes implemented in v4 will push them for many months, sending the message that reforming language, replacing terms that exclude and reinforce oppression in our systems is not so important to the “Cookie Autodelete” dev team.

**Comment by neroux** [comment]
Wholeheartedly agree with @Chaddock. A simple term which has been in use for years and isn't doing anything bad is "considered harmful". This is beyond reason. 🙄

The fact alone that someone seriously requested that and that it is actually being discussed is astonishing I am afraid. What do they come after next? Cookie? Delete?

**Comment by iam-py-test** [comment]
Why is it seem there has been no work on this?
Fixing https://github.com/Cookie-AutoDelete/Cookie-AutoDelete/issues/711 would make this even harder.
> This is beyond reason

Fixing this will actually make the language clearer, as _allowlist_ clearly means a list of allowed domains, while whitelist is extremely confusing until you either look it up or ask someone. Even though I know what it means, it still sounds like a list of white-colored objects.
It may take a small time to change for people used to this terminology, but in the end it will improve the UI and UX.
Also consider that everyone else is already switching, so still using _whitelist_ will make Cookie Auto Delete hard to use.

>  [the term whitelist promotes] the racist ideology of “white is good, black is bad”

I think the article posted by @AO-LocLab explains it better than I can

Thanks for your work on Cookie Auto Delete!

**Comment by neroux** [comment]
The terms white- and blacklist are industry standards and generally understood.

https://en.wiktionary.org/wiki/whitelist

There's no "confusion" and hence no need to change anything.

**Comment by iam-py-test** [comment]
> The terms white- and blacklist are industry standards and generall understood.
> 
> https://en.wiktionary.org/wiki/whitelist
> 
> There's no "confusion" and hence no need to change anything.

The terms `allowlist` and `denylist` are self-explanatory.
An `allowlist` is a list of allowed domains and a `denylist` is a list of domains denied access.
You always could use `blocklist` if that works better or sound better

**Comment by neroux** [comment]
That does not change what I wrote earlier. The currently used terms are industry standards and it is known what they stand for.

**Comment by iam-py-test** [comment]
> That does not change what I wrote earlier. The currently used terms are industry standards and it is known what they stand for.

What I meant is that `allowlist` & `denylist` are not hard to understand.
They are also becoming industry standards as major companies switch to using them

**Comment by neroux** [comment]
I don't think anybody here argued that they are hard to understand.

The point rather is that there is no need to change anything in the current setup.

And while they might not be _hard to understand_ they still sound awkward, simply because these have never been commonly used terms.

**Comment by iam-py-test** [comment]
> I don't think anybody here argued that they are hard to understand.
> 
> The point rather is that there is no need to change anything in the current setup.
> 
> And while they might not be _hard to understand_ they still sound awkward, simply because these have never been commonly used terms.

I agree that `denylist` sound extremally odd and hard to say, but I am sure that as a community we can determine something that works for everyone. 
I am going to stop posting any more comments on this thread as to avoid this getting too heated

**Comment by neroux** [comment]
There really is no reason to discuss such a change in the first place.

**Comment by bitboxer** [comment]
@neroux There are a lot. As pointed out above.

**Comment by kennethtran93** [comment]
Locking this issue before it gets heated even more.  Judging by the amount of reactions to the first post this isn't on top of the list of things to do at this point.

Denylist isn't the best to use because we are not denying cookies at all.  Allow would be fine, though would be non-existent once the below is integrated.

Plans for this includes the ability to have cookies check multiple lists, and lists to be self-configurable and re-nameable.

In addition the backend lists are also hardcoded as 'white' and 'grey' so changing those will take some time to allow users to properly migrate so that the configured lists still work (Kinda learnt that the hard way at one release point).  This isn't including the translations for each term.

**Comment by kennethtran93** [comment]
As a temporary solution to those that want it, I will attempt to add two settings - each will take in a string/text that allows you to 'visually' rename the lists.  The imports/exports will still be white or grey, but the visual look in the popup and list of expressions will have the text you put in.

Best of both worlds and a much easier solution to implement.

Unlocking for now, as locking it also prevented emojis/reactions from being used.

**Comment by neroux** [comment]
@bitboxer, not a single one I am afraid. Only arbitrarily motivated political agendas.

_I feel offended by the word cookie, please rename immediately_ 🙄

> Best of both worlds and a much easier solution to implement.

I am sorry, that is still caving in to ridiculous demands. Will you do the same if someone actually demands "cookie" to be renamed or any other string in your application?

**Comment by kennethtran93** [comment]
> Will you do the same if someone actually demands "cookie" to be renamed or any other string in your application?

If somebody wanted to rename cookie, then it would need a lot of visual changes from many browsers and websites, not just this web extension.

**Comment by neroux** [comment]
The question was solely in the context of CAD.

**Comment by iam-py-test** [comment]
For reference, a discussion about this for PiHole: https://discourse.pi-hole.net/t/change-blacklist-and-whitelist-terminology/31657/23

**Comment by ghost** [comment]
> It does not hurt you to use "Allowlist" instead. There is no harm done to you.

"They're only words, not using them can't hurt you", is right up there with, "they're only words, using them can't hurt you". 


> Hurtful language needs to stop. Language can change. To the better. 

"All usage of that word looks the same to me", goes nicely with "they all look the same to me" and "all black people are X".


In your effort to stamp out the pain of bigotry, you've become a bigot, punishing each and every individual for the actions of the racists among us, and employing the same illogical justifications as bigots always do.

**Comment by AderitoSilva** [comment]
Instead of using the concept of lists, I would suggest the concept of events. For example, instead of saying a site is on the _white_ or _gray_ list, I would prefer a site had an _auto-clean_ mode of “**On Domain Exited**”, “**On Tab Closed**”, “**On Session Ended**”, or “**Never**”. On the UI, we could then set this for a site using an option-cicler button with an optional dropdown button. Also, a tooltip on the button, describing what the hovered _auto-clean_ mode represents, would be more user-friendly/inclusive.

I suggest this, not because of cultural concerns, but because it looks more clear about the functionality we are configuring. The concept of _white_ and _gray_ lists doesn't seem clear enough to me. When I first started using CAD, I had to read the documentation to understand what does it mean for a site to be on the _gray_ list, for example. I recommended CAD for people I know, but they found it confusing and stop using it. They said they didn't like it, but I'm sure they just didn't understand how it works.

**Comment by neroux** [comment]
@AderitoSilva, I'd argue it should be pretty clear what _whitelist_ should mean/do. A list of exceptions where a certain functionality is not applied. As far as greylist is concerned, I do agree its meaning could be a bit ambiguous.

Having said all that, I do believe your suggestion is a viable "compromise" as it does not try to pursue an agenda and shift things around or no reason, but it's more of a feature enhancement in general. As I understand it even adds functionality which is currently not present. But I also believe this should warrant its own issue entry here.

**Comment by gitgrub** [comment]
I like the idea of @AderitoSilva, as it tells what the function actually does.
Changing the words white/black to something different will not change peoples mind.
(kidding a bit: looking at a traffic light, red for "stop" hurts Russians, yellow for "take care" hurts Asien ppl, green for "go" might make vegans happy :-)

Big companies doing the changes now, want to sell something, I don't think its changing their mindset.
Making the strings configurable is more work and might bring more confusion.
I think it would be better to put development time into the desired function of CAD.

Black could be understood as night, white as day, not necessarily what we are talking about here.

**Comment by neroux** [comment]
@gitgrub, finally we are onto a reasonable discussion 🙂

**Comment by kennethtran93** [comment]
Apologies for the rather long delay.  Initial commit has been changed to simple language term adjustments in linked PR.  I will need some time to propagate the adjustments into the actual codebase.

TL;DR:
- WhiteList -> KeepList
- GreyList -> RestartList
- List of Expressions -> List of Exceptions

**Comment by neroux** [comment]
All right, your decision of course, will switch to https://addons.mozilla.org/en-GB/firefox/addon/forget_me_not/ though

Not only did you not take @AderitoSilva's suggestion into account, you even ignored your own idea of making the name customisable but went the Full Monty and continued wiping out perfectly acceptable words to cater to some political idealogies.

Again, your decision, but also mine to switch to https://addons.mozilla.org/en-GB/firefox/addon/forget_me_not/

**Comment by kennethtran93** [comment]
It's not that I don't want to take @AderitoSilva's suggestion into account, but the current configuration doesn't easily allow it as some of the list setting is a global option (e.g. enable clean on domain change which is an additional trigger option in addition to the default clean on tab close).  It would take some additional design work to implement additional lists.

With Manifest V3 webextension on the horizon, and Bootstrap v5 already out, the majority of the codebase may need to be rewritten anyway.

**Comment by neroux** [comment]
Fair enough, as I [wrote](https://github.com/Cookie-AutoDelete/Cookie-AutoDelete/issues/754#issuecomment-907739268) it is more of a enhancement than anything else

> it's more of a feature enhancement in general. As I understand it even adds functionality which is currently not present. But I also believe this should warrant its own issue entry here.

so I understand that you did not go that route for the time being (e.g. upcoming interface version) but what I do not understand or agree with is that surrendering to that peer-pressure ideology. The "Expression List" could be possibly argued, but apart from that there was nothing wrong with these words and this whole issue is not motivated by making the extension better but to push a certain ideology and peer pressure people into following it. I would have also found the customisable names silly, but that would have given people at least a choice.

For what it's worth, your extension has not been the first that was replaced with an alternative in this context.

**Comment by bitboxer** [comment]
Thank you for changing this. Let me say thank you for doing this for a lot of people who will be super happy for this positive change ❤️ .

**Comment by neroux** [comment]
Congrats on _getting your way_ and _pushing your agenda_ 🙄

**Comment by mcquiggd** [comment]
Ridiculous.

**Comment by neroux** [comment]
Thanks for the suggestion, @mcquiggd

**Comment by ghost** [comment]
> In the face of all the current things going on in the world it would be a nice way of showing support by renaming graylist and whitelist to something else.

Colors and shades have always existed. Changing names doesn't change mentalities. Graylist means neither black nor white. Where's the problem?
What some people do out of elements of life such as colors is their problem. IMO changing names is a way of surrendering to their insane correlations between a color and a value. if we carry on this way black, white, gray, brown, yellow, red will one day all be banished on the account they implicitly nourish segregation based on the color of the skin.

A famous French singer, all in the absurdities of a certain French culture which perceives racism everywhere, had stated on a TV show that one day we'd all be blue... which supposes that removing differences removes segregation. Of course no segregation among robots. Should we become robots for the sake of egalitarianism? Long live differences, pluralism which improves all aspects of life. It's not my (our) problem if some poor souls bounce on differences to legitimate their hate speech, it's their problem, only.

**Comment by Youcue** [comment]
I downloaded the extension on Firefox for the first time today. I opened settings and saw "Enable Greylist Cleanup on Browser Restart". I've seen black and whitelist, but never "grey". I click on the question circle, sends me to the top of the documentation page on the wiki, even though it has the "#enable-greylist-cleanup-on-browser-restart" specified in the URL.

Ctrl + F, "greylist": Nothing. Thought I misspelled it, so double-checked the option and tried again, still nothing. Had to go back and look at the wording to guess that the name was updated to "RestartList" in the docs.

Naming was confusing before, even more confusing now, especially with docs updated before the app wording. Good change, I'm sure it'll be worth the effort.

**Comment by elandorr** [comment]
I found this accidentally when someone was confused by 'restartlist' and asked me.

White/gray/blacklist make most sense, even to outsiders. Gray clearly signals, 'not either white or black'. It's context-sensitive and universal. E.g. for e-mails, graylisting has nothing to do with restarting. You always need context. 'restart' has no meaning without specifying, either. A browser restart may not be significant. Today, browser uptime usually equals the system's.
White and black are universally known in the North and the South, that's not up for discussion at all. An artificial attempt to justify Zeitgeist abuse. It's used in India, too. It's mostly rich, white, spoiled kids who push this for their own egotism.

The arguments by the agenda pushers show how evil and senseless they are.

- Ban the word 'cookie', it offends gluttons.
- 'pihole'? Clearly a play on 'pie', gluttons will open their pie-holes and yell.
- 'delete'? Now, that's a controversial word, might as well say 'kill'. Oh, wait, we already do that all over computing. *Killing processes with a pipe*. Now I never thought of anything bad there, did you? Says a lot about you.
- 'Welcome'? The most dangerous of all. Through all of history, whenever a cute little tribe welcomed foreigners, they were abused and wiped out. At the very least, their culture shattered, their bloodlines broken. Obviously, 'welcome' has extremely negative connotations. Well, the dead don't complain much.
The usual agenda: Destroy the resistance, appeal to authority, eventually there'll be no one left who could resist. Fake smile until they come for you, too. Every useful fool eventually ends up a target himself.
![image](https://github.com/Cookie-AutoDelete/Cookie-AutoDelete/assets/56206745/29172c05-9c1f-4c3c-a074-15b0e9b4ab17)
Please remove this offensive term immediately! I demand it! The power of bigtech compels you!

The appeal to authority is more than dangerous. Because big tech plays on agenda, you follow them. Terrific argument. Big tech also boasts with killing people en masse, spreads lies for dystopian governments, and exploits everything to the last cent. Just like this agenda, all by and for money. The corpos you get your 'inspiration' from belong to the military-industrial complex and don't even hide it. Only deliberately complicit, or unbelievably ignorant parrots can worship them.

I, for one, will continue using CAD, as it's a frickin' extension with a simple task. As long as Mozilla doesn't ruin it. For a while, CAD was useless as it didn't clean up everything due to some interna. Monitoring such real issues in great detail is more worthy of time spent. Ignore the agenda, focus on the task. There are infinite ways for the internet to be abusive, be it fancy CSS exploits, localstorage shenanigans, or wannabe agenda warlords.
Mozilla constantly ruins things, so for CAD to serve a purpose, all it has to do is monitor efficacy. I wouldn't even care if the couple of buttons were in Chinese. You set it up once, then you remember where to click.
